BeeJet.Bot/Commands/SlashCommandContext.cs

```diff
@@ -0,0 +1,45 @@
+﻿using BeeJet.Bot.Extensions;
+using Discord;
+using Discord.WebSocket;
+
+namespace BeeJet.Bot.Commands
+{
+    public class SlashCommandContext : IBotReponseContext
+    {
+        public ISlashCommandInteraction SlashCommandInteraction { get; }
+
+        public IGuild Guild { get; set; }
```

Comment:
Deze properties zie ik vaker terugkomen. Zou dit dan niet in een basis klasse moeten waar de SlashCommandContext en de ButtonCommandContext van afleiden?
Want het enige wat veranders is hoe je de properties vult bij de Initialize.

Reply:
Die properties komen uit: IBotResponseContext
Ik dacht datg voor unit testing het handigste was als alles een interface was?
Kan er ook een baseobject van maken, heb nodig dat die een basisklasse, interface zijn, zodat straks services gebruik kunnen maken van een context, zonder dat ze nodig is dat ze kennis hebben of het een Button klik is of een slashcommand waar ze op reageren. Ze weten alleen wat de guild, user, client is. Want daar kun je eigenlijk alles al mee doen.

Reply:
Dat was ook mijn idee een beetje.
Dus een basis klasse welke de properties al heeft (want nu moet je ze in elke context definiëren),
En een interface om ze te sturen vanuit Unit Test en mocht je dit inderdaad nodig hebben voor je services.

Voor unit tests hoeft het niet per sé een Interface te zijn, al is dit wel het makkelijste (geeft ook de meeste flexibiliteit).
Het mag ook een abstracte klasse zijn (die methodes en properties kun je ook substituten)
Of in het uiterste geval een Virtual methode (is ook nog te substituten)

Reply:
Aangepast